src/index.js

```diff
@@ -1,40 +1,42 @@
 var _      = require("lodash"),
     assert = require("assert-plus");
 
-var RequestInterceptor = function(request, interceptors) {
-  if (!(this instanceof RequestInterceptor)) {
-    return new RequestInterceptor(request, interceptors);
+var RequestMiddlewareFramework = function(request) {
+  if (!(this instanceof RequestMiddlewareFramework)) {
+    return new RequestMiddlewareFramework(request);
   }
 
   assert.func(request, "request");
   this.request = request;
 
-  this.initialInterceptor = function(options, callback) {
+  this.initialMiddleware = function(options, callback) {
     request(options.uri, options, callback);
   };
 
-  this.interceptors = [ ];
+  this.middleware = [ ];
 
-  if (interceptors) {
-    this.use(interceptors);
+  if (arguments.length > 1) {
+    for (var i = 1; i < arguments.length; i++) {
```

Comment:
I don't want to do that for the first argument (since it's the request instance), and arguments isn't an array so I couldn't use shift(). But you're right there might be a better way to do this.